CvPoint[] points
float64 position_x
float64 position_z
float64 target_direction